packages/hasura-storage-js/src/machines/file-upload.ts

```diff
@@ -0,0 +1,174 @@
+      }
+    },
+    {
+      guards: {
+        hasFile: (context, event) => !!context.file || !!event.file
+      },
+
+      actions: {
+        incrementProgress: assign({
+          loaded: (_, { loaded }) => loaded,
+          progress: (_, { progress }) => progress
+        }),
+        setFileMetadata: assign({
+          id: (_, { id }) => id,
+          bucketId: (_, { bucketId }) => bucketId,
+          progress: (_) => 100
+        }),
+        setError: assign({ error: (_, { error }) => error }),
+        sendProgress: () => {},
+        sendError: () => {},
+        sendDestroy: () => {},
+        sendDone: () => {},
+        resetProgress: assign({ progress: (_) => null, loaded: (_) => 0 }),
+        addFile: assign({
+          file: (_, { file }) => file,
+          bucketId: (_, { bucketId }) => bucketId,
+          id: (_, { id }) => id
+        })
+      },
+      services: {
+        uploadFile: (context, event) => (callback) => {
+          const headers: AxiosRequestHeaders = {
+            'Content-Type': 'multipart/form-data'
+          }
+          const fileId = event.id || context.id
+          if (fileId) {
+            headers['x-nhost-file-id'] = fileId
+          }
+          const bucketId = event.bucketId || context.bucketId
+          if (bucketId) {
+            headers['x-nhost-bucket-id'] = bucketId
+          }
+          const name = event.name || context.file?.name
+          if (name) {
+            headers['x-nhost-file-name'] = name
+          }
+          const file = (event.file || context.file)!
+          const data = new FormData()
+          data.append('file', file)
+          if (event.adminSecret) {
+            headers['x-hasura-admin-secret'] = event.adminSecret
+          }
+          if (event.accessToken) {
+            headers['Authorization'] = `Bearer ${event.accessToken}`
+          }
+          let currentLoaded = 0
+          const controller = new AbortController()
+          axios
+            .post<{
+              bucketId: string
+              createdAt: string
+              etag: string
+              id: string
+              isUploaded: true
+              mimeType: string
+              name: string
+              size: number
+              updatedAt: string
+              uploadedByUserId: string
+            }>(event.url + '/files', data, {
+              headers,
+              signal: controller.signal,
+              onUploadProgress: (event: ProgressEvent) => {
+                const loaded = Math.round((event.loaded * file.size!) / event.total)
+                const additions = loaded - currentLoaded
+                currentLoaded = loaded
+                callback({
+                  type: 'UPLOAD_PROGRESS',
+                  progress: Math.round((loaded * 100) / event.total),
+                  loaded,
+                  additions
+                })
+              }
+            })
+            .then(({ data: { id, bucketId } }) => {
```

Comment:
Can you please convert this to async / await syntax?

Reply:
xstate callback services don't support async functions